test/transcoding/SPV_INTEL_fpga_dsp_control/prefer_dsp.ll

```diff
@@ -0,0 +1,99 @@
+; template<typename Function>
+; [[intel::prefer_dsp]]
+; void math_prefer_dsp_propagate(Function f)
+; {
+;   f();
+; }
+
+; int main() {
+;   math_prefer_dsp_propagate([]() {
+;     int a = 0;
+;     a += 1;
+;   });
+
+; return 0;
+; }
+
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv %t.bc --spirv-ext=+SPV_INTEL_fpga_dsp_control -o %t.spv
+; RUN: llvm-spirv %t.spv --to-text -o %t.spt
+; RUN: FileCheck < %t.spt %s --check-prefix=CHECK-SPIRV
+
+; RUN: llvm-spirv -r %t.spv -o %t.rev.bc
+; RUN: llvm-dis < %t.rev.bc | FileCheck %s --check-prefix=CHECK-LLVM
+
+; RUN: llvm-spirv %t.bc -o %t.negative.spv
+; RUN: llvm-spirv %t.negative.spv --to-text -o %t.negative.spt
+; RUN: FileCheck < %t.negative.spt %s --check-prefix=CHECK-SPIRV-NEG
+
+; RUN: llvm-spirv -r %t.negative.spv -o %t.negative.rev.bc
+; RUN: llvm-dis < %t.negative.rev.bc | FileCheck %s --check-prefix=CHECK-LLVM-NEG
+
+; CHECK-SPIRV: Capability FPGADSPControlINTEL
+; CHECK-SPIRV: Extension "SPV_INTEL_fpga_dsp_control"
+; CHECK-SPIRV: Name [[#REG1:]] "_Z25math_prefer_dsp_propagateIZ4mainE3$_0EvT_"
```

Comment:
@KornevNikita could you please use more meaningful names for variables, like `TargetFunction` instead of `REG1`?
There same comment for other `REG1`s and `REG2`s.